### PR TITLE
Support download file v2

### PR DESCRIPTION
**Pull request by justMaku**
Based on research by @bloerwald and documented at: https://wowdev.wiki/CASC#Download

### CODE REVIEW

File: CASCEdit/Handlers/DownloadHandler.cs [hunk 1]
```diff
@@ -50,18 +50,32 @@ public DownloadHandler(BLTEStream blte)
...
+					Header.NumFlags = br.ReadByte();
+				}
+
 				// entries
 				for (int i = 0; i < Header.NumEntries; i++)
 				{
 					var entry = new DownloadEntry()
 					{
-						Unknown = Header.Version > 1 ? br.ReadByte() : (byte)0, // new V2 field
 						Hash = new MD5Hash(br),
```
Comment (bloerwald) [inline CASCEdit/Handlers/DownloadHandler.cs]: Probably should get an `assert header.hash_size_ekey == MD5Hash::digest_size`.

File: CASCEdit/Handlers/DownloadHandler.cs [hunk 2]
```diff
@@ -50,18 +50,32 @@ public DownloadHandler(BLTEStream blte)
...
-						Stage = br.ReadByte(),
-						UnknownData = br.ReadBytes(4)
+						Stage = br.ReadByte()
 					};
 
+					if (Header.Unknown != 0) {
+						entry.Unknown = br.ReadUInt32BE();
+					}
+
+					if (Header.Version > 1) {
```
Comment (bloerwald) [inline CASCEdit/Handlers/DownloadHandler.cs]: Personal style: This is one of the rare cases where I prefer `≥ n` over `> n-1`.

File: CASCEdit/Handlers/DownloadHandler.cs [hunk 3]
```diff
@@ -100,7 +114,7 @@ public void AddEntry(CASResult blte)
 			{
 				Hash = blte.Hash,
 				FileSize = blte.CompressedSize - 30,
-				UnknownData = new byte[4],
+				Flags = new byte[Header.NumFlags],
```
Comment (bloerwald) [inline CASCEdit/Handlers/DownloadHandler.cs]: This is version dependent. Is this correctly initialised/degenerate nicely (i.e. empty) for header.version < 2?

**Comment by justMaku** [comment]
Please don't merge this until I address suggestions from @bloerwald.

**Comment by barncastle** [comment]
Good job both, this all looks fine to me. @bloerwald's suggestions would be ideal but I've not bothered with much redundancy or degradation anyway (refactoring and validation was stage 2 but I've lost interest) so it's up to you.

I've added you both as collaborators.

File: CASCEdit/Handlers/DownloadHandler.cs (hunk 3 above)
Comment (justMaku) [reply CASCEdit/Handlers/DownloadHandler.cs]: Currently it's impossible to create a new entity of this class without using an existing root manifest as a template.

**Comment by justMaku** [comment]
@bloerwald I've changed the version checks, other comments I will address in separate PRs.

**Pull request merged**